interop_binaries/src/bin/janus_interop_aggregator.rs

```diff
@@ -126,61 +128,43 @@ fn make_filter(
         },
     )?;
 
-    // Respect dap_serving_prefix.
-    let dap_filter = dap_serving_prefix
-        .split('/')
-        .filter_map(|s| (!s.is_empty()).then(|| warp::path(s.to_owned()).boxed()))
-        .reduce(|x, y| x.and(y).boxed())
-        .unwrap_or_else(|| warp::any().boxed())
-        .and(dap_filter);
-
-    let ready_filter = warp::path!("ready").map(|| {
-        warp::reply::with_status(warp::reply::json(&serde_json::json!({})), StatusCode::OK)
-            .into_response()
-    });
-    let endpoint_filter = warp::path!("endpoint_for_task").map(move || {
-        warp::reply::with_status(
-            warp::reply::json(&EndpointResponse {
+    let handler = Router::new()
+        .all(format!("{dap_serving_prefix}/*"), dap_handler)
+        .post("internal/test/ready", Json(serde_json::Map::new()))
+        .post(
+            "internal/test/endpoint_for_task",
+            Json(EndpointResponse {
                 status: "success",
-                endpoint: &dap_serving_prefix,
+                endpoint: dap_serving_prefix,
             }),
-            StatusCode::OK,
         )
-        .into_response()
-    });
-    let add_task_filter = warp::path!("add_task").and(warp::body::json()).then({
-        let datastore = Arc::clone(&datastore);
-        move |request: AggregatorAddTaskRequest| {
-            let datastore = Arc::clone(&datastore);
-            let keyring = Arc::clone(&keyring);
-            async move {
-                let response = match handle_add_task(&datastore, &keyring, request).await {
-                    Ok(()) => AddTaskResponse {
-                        status: SUCCESS.to_string(),
-                        error: None,
-                    },
-                    Err(e) => AddTaskResponse {
-                        status: ERROR.to_string(),
-                        error: Some(format!("{e:?}")),
+        .post(
+            "internal/test/add_task",
+            (
+                State(datastore),
```

Comment:
In this case it may be more readable just to move a clone of the datastore and keyring into the closure. Putting stuff in state is primarily useful when it would be otherwise onerous to wire it throughout the app in a bunch of places. Coincidentally moving a clone would slightly more efficient as well, but the primary consideration here is readability